### PR TITLE
Bugfix- Add manual payment modal

### PR DESCRIPTION
**Pull request by onkar-saeloun**
What-
- PR includes fixing the font size and padding issue on add payment modal
- PR includes implementation of new design on add payment modal
- Fix - Outside click on calendar should close calendar
- add symbol in front of amount

why-
- The new UI textbox and select box needed to be implemented
- The invoice dropdown was having UI issues on large names
- Calendar was not closing clicking outside
- amount symbol was not present

Notion-
https://www.notion.so/saeloun/Fix-the-UI-issues-on-Add-manual-payment-entry-modal-on-web-f2f8e82bffed4e3f8cdba0472250fd9f?pvs=4

loom-
https://www.loom.com/share/598fb2d7246540ffa9caedb192768473

### CODE REVIEW

File: app/javascript/src/components/payments/Modals/AddManualEntry.tsx [hunk 1]
```diff
@@ -188,128 +216,126 @@ const AddManualEntry = ({
...
-                      components={{
-                        Option: CustomOption,
-                        DropdownIndicator,
-                        IndicatorSeparator: () => null,
-                      }}
-                      onChange={handleInvoiceSelect}
-                      onMenuClose={() => setIsOpen(false)}
-                      onMenuOpen={() => setIsOpen(true)}
-                    />
+                    <div
```
Comment (onkar-saeloun) [inline app/javascript/src/components/payments/Modals/AddManualEntry.tsx:221]: replaced old select and input fields by new common components for uniform UI appearance

File: app/javascript/src/components/payments/Modals/AddManualEntry.tsx [hunk 2]
```diff
@@ -188,128 +216,126 @@ const AddManualEntry = ({
...
+                onClick={() =>
+                  setShowDatePicker({ visibility: !showDatePicker.visibility })
+                }
+              >
+                <CustomInputText
+                  disabled
+                  dataCy="transaction_date"
+                  id="transaction_date"
+                  label="Transaction Date"
+                  name="transaction_date"
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/payments/Modals/AddManualEntry.tsx:281]: dito
name should be camelCase

File: app/javascript/src/components/payments/Modals/AddManualEntry.tsx [hunk 3]
```diff
@@ -188,128 +216,126 @@ const AddManualEntry = ({
...
-                    ))}
-                  </select>
-                </div>
-              </div>
+            <div className="relative mt-4">
+              <CustomReactSelect
+                isSearchable
+                handleOnChange={e => setTransactionType(e.value)}
+                label="Transaction Type"
+                name="transaction-type"
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/payments/Modals/AddManualEntry.tsx:306]: dito
name should be camelCase

File: app/javascript/src/components/payments/Modals/AddManualEntry.tsx [hunk 4]
```diff
@@ -188,128 +216,126 @@ const AddManualEntry = ({
...
-                </div>
-              </div>
+              <CustomInputText
+                disabled
+                dataCy="payment-amount"
+                id="payment_amount"
+                inputBoxClassName="form__input block w-full appearance-none bg-white p-4 text-base h-12 focus-within:border-miru-han-purple-1000 border-miru-gray-1000"
+                label="Payment amount"
+                labelClassName="absolute top-0.5 left-1 h-6 z-1 origin-0 bg-white p-2 text-base font-medium duration-300 text-miru-dark-purple-200"
+                name="payment_amount"
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/payments/Modals/AddManualEntry.tsx:321]: dito
name should be camelCase

File: app/javascript/src/components/payments/Modals/AddManualEntry.tsx [hunk 5]
```diff
@@ -188,128 +216,126 @@ const AddManualEntry = ({
...
-                    rows={3}
-                    onChange={e => setNote(e.target.value)}
-                  />
-                </div>
-              </div>
+              <CustomTextareaAutosize
+                id="Notes-optional"
+                label="Notes (optional)"
+                maxRows={12}
+                name="Notes-optional"
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/payments/Modals/AddManualEntry.tsx:334]: dito
name should be camelCase

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

File: app/javascript/src/components/payments/Modals/AddManualEntry.tsx [hunk 6]
```diff
@@ -188,128 +216,126 @@ const AddManualEntry = ({
...
+                    >
+                      <CustomReactSelect
+                        ignoreDisabledFontColor
+                        isDisabled
+                        isSearchable
+                        classNamePrefix="border-0 font-medium text-miru-dark-purple-1000"
+                        defaultValue={invoice}
+                        handleOnChange={handleInvoiceSelect}
+                        label="Invoice"
+                        name="invoice-search"
```
[the remark this answers is not in the document]
Comment (onkar-saeloun) [reply app/javascript/src/components/payments/Modals/AddManualEntry.tsx]: Done

**Review by prasanthchaduvula — APPROVED** [review]

**Pull request merged**